slack/web/base_client.py

```diff
@@ -265,7 +265,8 @@ def _get_url(self, api_method):
                 )
             response = {"data": data, "headers": res.headers, "status_code": res.status}
 
-        await session.close()
+        if not self.session or self.session.closed:
```

Comment:
If I understand correctly, does the code you intended seem to be 2) in the following examples?

```python
class Session:
  def closed(self):
    true

null_session = None
closed_session = Session()

# 1)
if not null_session or closed_session.closed:
  print("x")

# 2)
if not (null_session or closed_session.closed):
  print("y")
```

Result:
```
>>> class Session:
...   def closed(self):
...     true
... 
>>> null_session = None
>>> closed_session = Session()
>>> 
>>> 
>>> if not null_session or closed_session.closed:
...   print("x")
... 
x
>>> 
>>> if not (null_session or closed_session.closed):
...   print("y")
... 
>>> 
```

Reply:
Thanks for the further explanation. I understand the point.

To me, double negative logic looks a bit confusing. Defining a local value may be easier-to-understand (and is good for maintainability, too).

```python
use_running_session = self.session and not self.session.closed
if use_running_session:
    session = self.session
else:
    session = aiohttp.ClientSession(
        timeout=aiohttp.ClientTimeout(total=self.timeout),
        auth=req_args.pop("auth", None),
    )

(omitted)

if not use_running_session:
  await session.close()
```

Reply:
You're right, this approach looks better.
Thanks for your input, I've updated the commit.